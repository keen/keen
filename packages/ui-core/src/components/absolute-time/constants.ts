export const DATE_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
