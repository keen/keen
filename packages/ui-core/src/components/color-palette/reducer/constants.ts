export const SET_IS_DRAGGED = 'SET_IS_DRAGGED';
export const SET_COLORS = 'SET_COLORS';
export const SET_ACTIVE_COLOR = 'SET_ACTIVE_COLOR';
export const SET_ADD_COLOR_PICKER_OPEN = 'SET_ADD_COLOR_PICKER_OPEN';
export const SET_COLOR_PICKER_POSITION = 'SET_COLOR_PICKER_POSITION';
