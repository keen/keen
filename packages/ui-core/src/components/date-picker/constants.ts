export const DATE_FORMAT = 'YYYY-MM-DD';
export const TIME_FORMAT = 'HH:mm';
