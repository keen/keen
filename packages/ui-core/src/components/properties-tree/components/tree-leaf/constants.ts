export const SEPARATOR = '.';
export const PADDING = 15;
