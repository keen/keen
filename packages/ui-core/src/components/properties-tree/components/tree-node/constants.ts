export const PADDING = 15;
