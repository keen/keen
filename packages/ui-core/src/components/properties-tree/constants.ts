export const ELEMENT_HEIGHT = 36;
export const LIST_HEIGHT = 200;
