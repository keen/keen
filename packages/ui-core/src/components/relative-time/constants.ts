export const RELATIVITY_OPTIONS = [
  {
    label: 'this',
    value: 'this',
  },
  {
    label: 'previous',
    value: 'previous',
  },
];

export const THIS_RELATIVITY = 'this';
export const PREVIOUS_RELATIVITY = 'previous';
