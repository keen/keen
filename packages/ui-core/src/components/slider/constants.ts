export const SET_DIMENSION = 'SET_DIMENSION';
export const SET_VALUE = 'SET_VALUE';
export const SET_DRAG_POSITION = 'SET_DRAG_POSITION';
export const SET_DRAG_STATE = 'SET_DRAG_STATE';
export const SET_DRAG_MOVING = 'SET_DRAG_MOVING';

export const DRAG_CONTROL_ID = '@keen-drag-control';

export const TOOLTIP_CONTENT_MAX_WIDTH = 200;
