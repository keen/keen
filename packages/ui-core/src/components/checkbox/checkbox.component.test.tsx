import React, { ComponentProps } from 'react';
import { render as rtlRender, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import Checkbox from './checkbox.component';
import { KEYBOARD_KEYS } from '../../constants';

const render = (overProps: Partial<ComponentProps<typeof Checkbox>> = {}) => {
  const props = {
    id: '@checkbox-1',
    ...overProps,
  };

  const wrapper = rtlRender(<Checkbox {...props} />);

  return {
    wrapper,
    props,
  };
};

test('should call "onChange" handler', () => {
  const mockFn = jest.fn();
  const {
    wrapper: { container },
    props,
  } = render({ onChange: mockFn });

  const element = container.querySelector('input[type="checkbox"]');
  fireEvent.click(element);

  expect(props.onChange).toHaveBeenCalled();
});

test('should call "onChange" handler on Enter press', () => {
  const mockFn = jest.fn();
  const {
    wrapper: { container },
    props,
  } = render({ onChange: mockFn });

  const element = container.querySelector('input[type="checkbox"]');
  fireEvent.keyPress(element, { key: 'Enter', keyCode: KEYBOARD_KEYS.ENTER });

  expect(props.onChange).toHaveBeenCalled();
});

test('should set checkbox element "checked" to truthy value', () => {
  const {
    wrapper: { container },
  } = render({ checked: true });
  const element = container.querySelector(
    'input[type="checkbox"]'
  ) as HTMLInputElement;

  expect(element.checked).toBe(true);
});

test('should set checkbox element "checked" to false value', () => {
  const {
    wrapper: { container },
  } = render({ checked: false });
  const element = container.querySelector(
    'input[type="checkbox"]'
  ) as HTMLInputElement;

  expect(element.checked).toBe(false);
});

test('should set checkbox element "disabled" to truthy value', () => {
  const {
    wrapper: { container },
  } = render({ disabled: true });
  const element = container.querySelector(
    'input[type="checkbox"]'
  ) as HTMLInputElement;

  expect(element.disabled).toBe(true);
});

test('should set checkbox element "disabled" to false value', () => {
  const {
    wrapper: { container },
  } = render({ disabled: false });
  const element = container.querySelector(
    'input[type="checkbox"]'
  ) as HTMLInputElement;

  expect(element.disabled).toBe(false);
});

test('should set primary checkbox style', () => {
  const {
    wrapper: { container },
  } = render();
  expect(container).toMatchSnapshot();
});

test('should set secondary checkbox style', () => {
  const {
    wrapper: { container },
  } = render({ type: 'secondary' });
  expect(container).toMatchSnapshot();
});
