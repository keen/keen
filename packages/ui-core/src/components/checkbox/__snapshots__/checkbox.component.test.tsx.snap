// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set primary checkbox style 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  line-height: 1;
}

.c0:focus-visible {
  outline: 1px solid #85B4C3;
}

.c2 {
  position: relative;
  display: inline-block;
  box-sizing: content-box;
  border: solid 1px #27566D;
  border-radius: 2px;
  width: 12px;
  height: 12px;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div>
  <div
    class="c0"
    role="checkbox"
    tabindex="0"
  >
    <input
      class="c1"
      id="@checkbox-1"
      type="checkbox"
    />
    <div
      class="c2"
      type="primary"
    />
  </div>
</div>
`;

exports[`should set secondary checkbox style 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  line-height: 1;
}

.c0:focus-visible {
  outline: 1px solid #85B4C3;
}

.c2 {
  position: relative;
  display: inline-block;
  box-sizing: content-box;
  border: solid 1px rgba(39,86,109,0.2);
  background: rgba(119,163,187,0.05);
  border-radius: 4px;
  width: 15px;
  height: 15px;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<div>
  <div
    class="c0"
    role="checkbox"
    tabindex="0"
  >
    <input
      class="c1"
      id="@checkbox-1"
      type="checkbox"
    />
    <div
      class="c2"
      type="secondary"
    />
  </div>
</div>
`;
